src/lib.rs

```diff
@@ -365,4 +367,12 @@ pub trait StructOpt {
     fn from_args() -> Self where Self: Sized {
         Self::from_clap(&Self::clap().get_matches())
     }
+
+    /// Gets the struct from any iterator such as a `Vec` of your making.
+    /// Print the error message and quit the program in case of failure.
+    fn from_any<I, T>(iter: I) -> Self where
+        Self: Sized, I: IntoIterator<Item = T>, T: Into<OsString> + Clone,
```

Comment:
```rust
fn from_iter<I>(iter: I) -> Self
where
    Self: Sized,
    I: IntoIterator,
    I::Item: Into<OsString> + Clone
{
    //...
```

Reply:
That was not really about indentation, but about the name of the method and the useless generic.

Reply:
Ho ! I will update that, sorry, didn't see.

Reply:
done!